moveit_core/collision_detection/src/world.cpp

```diff
@@ -225,9 +225,14 @@ bool World::moveObject(const std::string& object_id, const Eigen::Isometry3d& tr
   if (transform.isApprox(Eigen::Isometry3d::Identity()))
     return true;  // object already at correct location
   ensureUnique(it->second);
-  for (size_t i = 0, n = it->second->shapes_.size(); i < n; ++i)
+  it->second->pose_ = transform * it->second->pose_;
+  for (Eigen::Isometry3d shape_pose : it->second->shape_poses_)
```

Comment:
The type should be Eigen::Isometry3d&. Right now this loop does not modify the shape pose.

Reply:
If you use `auto&` it will make it a reference and the compiler will make sure it is the same type and you don't get any unintended conversions.  Good catch @jliukkonen

Reply:
The reason I didn't initially recommend `auto` type is because of the warning against auto type here: [https://eigen.tuxfamily.org/dox/TopicPitfalls.html](https://eigen.tuxfamily.org/dox/TopicPitfalls.html). But reading that through now again I'm not entirely certain it applies here. Probably doesn't since there are no Eigen expressions involved...?

Reply:
I think there was a rough guideline for MoveIt to use range-based for loops, but only use auto when the variable is defined nearby. It is good not to have to look up the type while reading the code.

Reply:
You are right about the guidance for MoveIt.  This should be a reference to avoid a copy.